pkg/cloud/services/ec2/launchtemplate.go

```diff
@@ -45,18 +46,45 @@ func (s *Service) GetLaunchTemplate(name string) (*expinfrav1.AWSLaunchTemplate,
 	out, err := s.EC2Client.DescribeLaunchTemplateVersions(input)
 	switch {
 	case awserrors.IsNotFound(err):
-		return nil, nil
+		return nil, "", nil
 	case err != nil:
 		s.scope.Info("", "aerr", err.Error())
+		return nil, "", err
 	}
 
 	if len(out.LaunchTemplateVersions) == 0 {
-		return nil, nil
+		return nil, "", nil
 	}
 
 	return s.SDKToLaunchTemplate(out.LaunchTemplateVersions[0])
 }
 
+// GetLaunchTemplateId returns the existing LaunchTemplateId or empty string if it doesn't exist.
+func (s *Service) GetLaunchTemplateID(launchTemplateName string) (string, error) {
+	if launchTemplateName == "" {
+		return "", nil
+	}
+
+	input := &ec2.DescribeLaunchTemplateVersionsInput{
+		LaunchTemplateName: aws.String(launchTemplateName),
+		Versions:           aws.StringSlice([]string{expinfrav1.LaunchTemplateLatestVersion}),
+	}
+
+	out, err := s.EC2Client.DescribeLaunchTemplateVersions(input)
+	switch {
+	case awserrors.IsNotFound(err):
+		return "", nil
+	case err != nil:
+		s.scope.Info("", "aerr", err.Error())
```

Comment:
There's a missing `return` here (same as [here](https://github.com/newrelic-forks/cluster-api-provider-aws/pull/82/files#diff-34e3c64d496f11b3389c204875c80487e4861f29a93821922c1284086e0741fbR52)) but it's also missing in the [upstream PR](https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2401/files#diff-34e3c64d496f11b3389c204875c80487e4861f29a93821922c1284086e0741fbR83) so not fixing it for consistency.